### PR TITLE
Doing fresh clone for libdrm, mesa and minigbm for Travis CI

### PR DESCRIPTION
**Pull request by codernavi18**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -27,6 +34,30 @@ script:
...
+- git clone https://github.com/intel/external-libdrm /tmp/libdrm
+- pushd /tmp/libdrm
+- ./autogen.sh --disable-radeon --disable-nouveau --disable-amdgpu --enable-udev --enable-libkms --prefix=$WLD
+- make -j5 && make install
+- popd
+
+#clone and build mesa : dependency
+- git clone https://github.com/intel/external-mesa /tmp/mesa
+- pushd /tmp/mesa
+- ./autogen.sh --prefix=$WLD --enable-gles2 --disable-gallium-egl --with-egl-platforms=surfaceless --disable-dri3 --enable-shared-glapi --enable-texture-float --with-driver=dri --disable-glu --disable-glut --without-demos --disable-option-checking --disable-glx --enable-glx-tls --disable-nouveau --disable-gbm --with-gallium-drivers= --with-dri-drivers=i915,i965 --with-vulkan-drivers=intel
```
Comment (tpalli) [inline .travis.yml]: There are some old and deprecated options used here, where do these originate?

Following options should not be needed:

--enable-gles2
--disable-gallium-egl
--with-driver=dri
--disable-glu
--disable-glut
--without-demos
--disable-option-checking
--disable-nouveau

following options are deprecated:

 --with-egl-platforms=surfaceless (this should be "--with-platforms=surfaceless")

this is weird (you don't want glx but you want glx-tls?):

 --disable-glx --enable-glx-tls

following can be said in a prettier way:

--with-gallium-drivers= --with-dri-drivers=i915,i965

could look like: "--without-gallium-drivers --with-dri-drivers=i965" (we don't need i915 and we don't want gallium drivers)

Comment (kalyankondapally) [inline .travis.yml]: @codernavi18 Can u address the mentioned issues here ? Update our Google doc to build HWC on linux at the same time ?

**Comment by codernavi18** [comment]
@tpalli @kalyankondapally Made the suggested changes
@kalyankondapally Updated the google doc

**Comment by kalyankondapally** [comment]
Please include JIra and Test in u commit message from next time

**Comment by codernavi18** [comment]
@kalyankondapally  : Thanks. I will strictly take care for future commits.

**Pull request closed without merging**